cl/recap/tasks.py

```diff
@@ -2092,17 +2108,71 @@ def fetch_attachment_page(self: Task, fq_pk: int) -> None:
             "attachment data"
         )
         mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
-        return
+        self.request.chain = None
+        return []
     except RECAPDocument.DoesNotExist as exc:
         msg = "Could not find docket to associate with attachment metadata"
         if self.request.retries == self.max_retries:
             mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
-            return
+            self.request.chain = None
+            return []
         mark_fq_status(fq, msg, PROCESSING_STATUS.QUEUED_FOR_RETRY)
         raise self.retry(exc=exc)
     msg = "Successfully completed fetch and save."
     mark_fq_status(fq, msg, PROCESSING_STATUS.SUCCESSFUL)
 
+    # Logic to replicate the attachment page to sub-dockets matched by RECAPDocument
+    if is_appellate_court(court_id):
+        # Subdocket replication for appellate courts is currently not supported.
+        self.request.chain = None
+        return []
+
+    sub_docket_main_rds = list(
+        get_main_rds(court_id, pacer_doc_id).exclude(
+            docket_entry__docket__pacer_case_id=pacer_case_id
+        )
+    )
+    sub_docket_pqs = []
+    for main_rd in sub_docket_main_rds:
+        # Create PQs related to RD that require replication.
+        sub_docket_pqs.append(
+            ProcessingQueue(
+                uploader_id=fq.user_id,
+                pacer_doc_id=main_rd.pacer_doc_id,
+                pacer_case_id=main_rd.docket_entry.docket.pacer_case_id,
+                court_id=court_id,
+                upload_type=UPLOAD_TYPE.ATTACHMENT_PAGE,
+                filepath_local=ContentFile(
+                    text.encode(), name="attachment_page.html"
+                ),
+            )
+        )
+
+    if sub_docket_pqs:
+        pqs_created = ProcessingQueue.objects.bulk_create(sub_docket_pqs)
+        return [pq.pk for pq in pqs_created]
+
+    self.request.chain = None
+    return []
```

Comment:
I've noticed this block of code handles early exits differently than other parts of the task.  For consistency, I propose refactoring it to check the exit condition first, like this:

```suggestion
    if not sub_docket_pqs:
        self.request.chain = None
        return []

    pqs_created = ProcessingQueue.objects.bulk_create(sub_docket_pqs)
    return [pq.pk for pq in pqs_created]
```

This change makes the code consistent with the other early exit patterns